discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
+
+```json
+[
+  {
+    "provider": "Cloud Storage Provider",
+    "description": "Blob storage in the cloud",
+    "provider_uri": "https://cloud.example.com/docs/storage"
+  },
+  {
+    "provider": "Cloud MySQL"
+  },
+  {
+    "provider": "Cloud OtherSQL",
+    "description": "Highly scalable SQL service"
+  }
+]
+```
+
+#### `type` entity
+
+Used in discovery for enumerating the different CloudEvent `type`s represented
+in this discovery system.
+
+##### `type` entity attributes
+
+###### type
+
+- Type: `String`
+- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+  attribute.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string, following the constraints as defined in the
+    CloudEvents spec.
+- Examples:
+  - "com.github.pull.create"
+  - "com.example.object.delete.v2"
+
+###### specversion
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: CloudEvents [`specversion`](https://github.com/cloudevents/spec/blob/master/spec.md#specversion)
+  that will be used for events published for this producer, event type
+  combination.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+
+###### datacontenttype
+
+- Type: `String`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#datacontenttype)
+  attribute. Indicating how the `data` attribute of subscribed events will be
+  encoded.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+
+###### dataschema
+
+- Type: `URI`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#dataschema)
+  attribute. This identifies the canonical storage location of the schema of
+  the `data` attribute of subscribed events.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+
+###### dataschematype
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: If using `dataschemacontent` for inline schema storage, the
+  `dataschematype` indicates the type of schema represented there.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Examples:
+  - "application/json"
+  -
+
+###### dataschemacontent
+
+- Type: `String`
+- Description: An inline representation of the schema of the `data` attribute
+  encoding mechanism. This is an alternative to using the `dataschema`
+  attribute.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string containing a schema compatible with
+    the `datacontenttype`.
+  - If `dataschama` is present, this field MUST NOT be present.
+
+##### `type` entity examples
+
+A single `type` entity.
+
+```json
+{
+  "type": "com.example.storage.object.create",
+  "specversion": "1.0",
+  "datacontenttype", "application/json",
```

Comment:
Something I've been thinking about is whether implementations might want to support more than 1 content type/encoding for a single event type. For example, a subscriber might have tooling and services already oriented around a specific format. Additionally, subscribers might want to request a more compact representation if bandwidth/space is a concern. 

The same could possibly be said about the schema format, though that gets more complex to specify generally, as features across different schema technologies will vary.

Reply:
I think it's unlikely that systems would offer a variety of encodings of the data attribute on a per-subscriber basis. It would complicate their systems quite a bit.